src/constraints.jl

```diff
@@ -0,0 +1,645 @@
+    else
+        error(error_msg)
+    end
+    func = :($op($lhs, $rhs0))
+    parse_code = :()
+    build_code = :(JuMP.build_constraint($(_error), $(esc(func)), $set))
+    return parse_code, build_code
+end
+
+# --> cannot be used because it clashes with indicator constraint parsing in JuMP.
+# """
+#     function JuMP.parse_constraint_head(
+#         _error::Function,
+#         ::Val{:(-->)},
+#         lhs,
+#         rhs
+#     )
+
+# Extend `JuMP.parse_constraint_head` parse logical proposition constraints that
+# use the implication operator `-->`.
+# """
+# function JuMP.parse_constraint_head(
+#     _error::Function,
+#     ::Val{:(-->)},
+#     lhs,
+#     rhs
+# )
+#     error_msg = 
+#         "The MOI set in the logical constraint $lhs --> $rhs was not identified. " *
+#         "The set `MOI.EqualTo{Bool}(value)` should be preceded by `==` or `in`."
+#     if rhs.args[1] in (:in, :(==))
+#         rhs0 = rhs.args[2]
+#         set = rhs.args[3] isa Bool ? _MOI.EqualTo(rhs.args[3]) : 
+#                 rhs.args[3] isa _MOI.EqualTo ? rhs.args[3] : error(error_msg)
+#     else
+#         error(error_msg)
+#     end
+#     func = :(-->($lhs, $rhs0))
+#     parse_code = :()
+#     build_code = :(JuMP.build_constraint($(_error), $(esc(func)), $set))
+#     return false, parse_code, build_code
+# end
+
+"""
+    JuMP.build_constraint(
+        _error::Function, 
+        func::AbstractVector{<:Union{Number, LogicalVariableRef, _LogicalExpr}}, # allow any vector-like JuMP container
+        set::_MOI.AbstractVectorSet # Keep general to allow CP sets from MOI
+    )
+
+Extend `JuMP.build_constraint` to add logical cardinality constraints to a [`GDPModel`](@ref). 
+This in combination with `JuMP.add_constraint` enables the use of 
+`@constraint(model, [name], logical_expr in set)`, where set can be either of the following
+cardinality sets: `AtLeast(n)`, `AtMost(n)`, or `Exactly(n)`.
+
+## Example
+
+To select exactly 1 logical variable `Y` to be `true`, do 
+(the same can be done with `AtLeast(n)` and `AtMost(n)`):
+
+```jldoctest
+julia> model = GDPModel();
+julia> @variable(model, Y[i = 1:2], LogicalVariable);
+julia> @constraint(model, [Y[1], Y[2]] in Exactly(1));
+```
+
+    JuMP.build_constraint(
+        _error::Function, 
+        func::_LogicalExpr,
+        set::_MOI.EqualTo{Bool}
+    )
+
+Extend `JuMP.build_constraint` to add logical propositional constraints to a [`GDPModel`](@ref). 
+This in combination with `JuMP.add_constraint` enables the use of 
+`@constraint(model, [name], logical_expr == true/false)` to define a Boolean expression that must
+either be true or false.
+"""
+function JuMP.build_constraint(
+    _error::Function, 
+    func::AbstractVector{<:Union{Number, LogicalVariableRef, _LogicalExpr}}, # allow any vector-like JuMP container
+    set::_MOI.AbstractVectorSet # Keep general to allow CP sets from MOI
+    )
+    return JuMP.VectorConstraint(func, set)
+end
+
```

Comment:
We'll need to add dispatch methods for `_MOI.EqualTo` (not `Bool`) to check and convert to `Bool` where possible and then call the other `build_constraint` methods.

Reply:
done